lib/azure/armrest/storage/disk_service.rb

```diff
@@ -11,6 +11,115 @@ class DiskService < ResourceGroupBasedService
+        # how long the privately generated SAS token should last. This token
+        # is used internally by requests that are used to access the requested
+        # information. By default it lasts for 1 hour.
+        #
+        # Get the information you need using:
+        #
+        # * response.body    - blob data.
+        # * response.headers - blob metadata.
+        #
+        # Example:
+        #
+        #   vms = Azure::Armrest::VirtualMachineService.new(conf)
+        #   sds = Azure::Armrest::Storage::DiskService.new(conf)
+        #
+        #   vm = vms.get(vm_name, vm_resource_group)
+        #   os_disk = vm.properties.storage_profile.os_disk
+        #
+        #   disk_id = os_disk.managed_disk.id
+        #   disk = sds.get_by_id(disk_id)
+        #
+        #   # Get the first 1024 bytes
+        #   response = sds.get_blob_raw(disk.name, disk.resource_group, :range => 0..1023)
+        #
+        #   p response.headers
+        #   File.open('vm.vhd', 'a'){ |fh| fh.write(response.body) }
+        #
+        def get_blob_raw(disk_name, resource_group = configuration.resource_group, options = {})
+          validate_resource_group(resource_group)
+
+          options[:access] = 'read'            # Must be 'read' at the moment
+          options[:durationInSeconds] ||= 3600 # 1 hour default
+
+          # Must delete these before proceding to first request
+          range = options.delete(:range)
+          start_byte = options.delete(:start_byte)
+          end_byte = options.delete(:end_byte)
+          length = options.delete(:length)
+          entire_image = options.delete(:entire_image)
+
+          # This call will give us an operations URL in the headers.
+          initial_url = build_url(resource_group, disk_name, 'BeginGetAccess')
+          response = rest_post(initial_url, options.to_json)
+          headers = ResponseHeaders.new(response.headers)
+
+          # Using the URL returned from the above call, make another call that
+          # will return the URL + SAS token.
+          op_url = headers.try(:azure_asyncoperation) || headers.location
+
+          unless op_url
+            msg = "Unable to find an operations URL for #{disk_name}/#{resource_group}"
+            raise Azure::Armrest::NotFoundException.new(response.code, msg, response.body)
+          end
+
+          # Dig the URL + SAS token URL out of the response
+          response = rest_get(op_url)
+          body = ResponseBody.new(response.body)
+          sas_url = body.try(:properties).try(:output).try(:access_sas)
+
+          unless sas_url
+            msg = "Unable to find an SAS URL for #{disk_name}/#{resource_group}"
+            raise Azure::Armrest::NotFoundException.new(response.code, msg, response.body)
+          end
+
+          # The same restrictions that apply to the StorageAccont method also apply here.
+          range_str = nil
+
+          if range
+            range_str = "bytes=#{range.min}-#{range.max}"
```

Comment:
You can have logic to create a `range` object based on all combinations for start, end, length etc, and convert to `range_str` with one shot.